### PR TITLE
Enable Affliction for DF

### PR DESCRIPTION
**Pull request by jharrell**
Hi there! First PR. I believe I got this mostly right (was able to load an old report locally). Appreciate any and all feedback!

### CODE REVIEW

File: src/analysis/retail/warlock/affliction/constants.ts [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
-  [SPELLS.IMPENDING_CATASTROPHE_DEBUFF.id]: 12000,
+  [talents.SIPHON_LIFE_TALENT.id]: 15000,
+  [talents.UNSTABLE_AFFLICTION_TALENT.id]: 16000,
+  [talents.PHANTOM_SINGULARITY_TALENT.id]: 16000,
+  [talents.VILE_TAINT_TALENT.id]: 10000,
+  [talents.SOUL_ROT_TALENT.id]: 8000,
+  // Shadowlands Covenant Abilities
+  [covenantSpells.SCOURING_TITHE.id]: 18000,
+  [covenantSpells.SOUL_ROT.id]: 8000,
+  [covenantSpells.IMPENDING_CATASTROPHE_DEBUFF.id]: 12000,
```
Comment (emallson) [inline src/analysis/retail/warlock/affliction/constants.ts:19]: you don't have to, but commenting that it *is* okay to delete these if they're not present in the talent trees

File: src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx [hunk 2]
```diff
@@ -38,39 +39,9 @@ class AlwaysBeCasting extends CoreAlwaysBeCasting {
...
-            </ul>
-          </>
-        }
-      >
-        <div className="pad">
-          <label>Active time</label>
-
-          <Gauge value={this.activeTimePercentage} />
-        </div>
-      </Statistic>
```
Comment (emallson) [inline src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx]: what is the motivation behind removing this? was it causing errors? or is this now a part of the superclass and not needed here?

**Review by emallson — COMMENTED** [review]
small comment about a removed statistic, otherwise this looks good

File: src/analysis/retail/warlock/affliction/constants.ts (hunk 1 above)
Comment (jharrell) [reply src/analysis/retail/warlock/affliction/constants.ts:19]: Sounds good! Yeah I think I'm going to have multiple PRs bringing this spec in line with some others, so I'll tackle a cleanup of SL abilities then, if that's ok?

File: src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx (hunk 2 above)
Comment (jharrell) [reply src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx]: I picked a spec at random (fury) and saw it didn't exist. I see that it exists in a couple of others so I'll get this added back. Thanks!

File: src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx [hunk 3]
```diff
@@ -38,16 +40,18 @@ class AlwaysBeCasting extends CoreAlwaysBeCasting {
...
   }
 
   statistic() {
     return (
       <Statistic
-        position={STATISTIC_ORDER.CORE(2)}
+        position={STATISTIC_ORDER.CORE(10)}
         tooltip={
-          <>
+          <Trans id="warlock.affliction.alwaysBeCasting.statistic.tooltip">
```
Comment (ToppleTheNun) [inline src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx:54]: Just a note that I don't see this string in any of the localization files.

File: src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx [hunk 4]
```diff
@@ -62,12 +66,13 @@ class AlwaysBeCasting extends CoreAlwaysBeCasting {
...
             </ul>
-          </>
+          </Trans>
         }
       >
         <div className="pad">
-          <label>Active time</label>
-
+          <label>
+            <Trans id="warlock.affliction.alwaysBeCasting.statistic.label">Active time</Trans>
```
Comment (ToppleTheNun) [inline src/analysis/retail/warlock/affliction/modules/features/AlwaysBeCasting.tsx:74]: I don't see this one either.

**Review by ToppleTheNun — APPROVED** [review]
Just notes on localization messages.

**Pull request merged**